### PR TITLE
docs: Add a page about public API changes in dnf 5.2.0.0

### PR DESCRIPTION
**Pull request by jan-kolarik**
To better communicate the changes introduced with the new major version.

This will be then also referenced on the 5.2.0.0 GitHub release page.

### CODE REVIEW

File: doc/api/changes_dnf5.2.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+
+Wrapping struct attributes
+==========================
+
+Public attributes in struct objects have been wrapped in getter and setter methods.
+
+Before: ``bool ignore_case;`` Now: ``void set_ignore_case(bool); bool get_ignore_case() const;``
+
+* ``libdnf5::GoalJobSettings``
```
Comment (bcl) [inline doc/api/changes_dnf5.2.rst:15]: It isn't immediately obvious what all this encompasses. For example, when `group_with_name` stops working I search the changelog for mention of it, not of GoalJobSetting.

Detailing all the changes would help when searching.

File: doc/api/changes_dnf5.2.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+* ``libdnf5::comps::GroupSack``
+* ``libdnf5::repo::Repo::fresh()`` (use ``is_expired()`` instead)
+
+
+Other changes
+=============
+
+* ``libdnf5::advisory::AdvisoryQuery``
+
+  * ``filter_*`` methods have the default ``cmp_type`` ``IEXACT`` (was ``EQ`` before)
```
Comment (gotmax23) [inline doc/api/changes_dnf5.2.rst:54]: What's the difference between `EQ` and `IEXACT`? Is it just that `IEXACT` is case-insensitive? What was the rationale for this change?

Comment (gotmax23) [reply doc/api/changes_dnf5.2.rst:54]: Hmm, I still see `EQ` as the default in

https://github.com/rpm-software-management/dnf5/blob/1fe03c783b4bba7288eb4aa795718199102caa31/include/libdnf5/rpm/package_query.hpp#L83

and elsewhere. Am I missing something?

Comment (pkratoch) [reply doc/api/changes_dnf5.2.rst:54]: Yes, `IEXACT` is just case insensitive. It's not all filter methods, just `filter_severity` and `filter_type`. The reason is that these two accept only specified values and it shouldn't matter if e.g. the type is "Bugfix" or "bugfix". See the commit: https://github.com/rpm-software-management/dnf5/commit/355aef18061bbee5599d70bd07da71d8db07c318

Comment (jan-kolarik) [reply doc/api/changes_dnf5.2.rst:54]: Also, the `IEXACT` default is related to the mentioned `libdnf5::advisory::AdvisoryQuery` object. You are referencing the `libdnf5::rpm::PackageQuery` object which should be unchanged.

File: doc/api/changes_dnf5.2.rst (hunk 1 above)
Comment (jan-kolarik) [reply doc/api/changes_dnf5.2.rst:17]: I've added a file and linked it to the docs with diff between dnf5-5.1.17 and dnf5-5.2.0.0 API changes. I guess that would be better for the text search.

File: doc/api/changes_dnf5.2.rst (hunk 2 above)
Comment (jan-kolarik) [reply doc/api/changes_dnf5.2.rst]: But yeah, I've used `filter_*`, although just some methods were affected by this. Fixed now.

**Comment by jan-kolarik** [comment]
@bcl @gotmax23 Please check again. I've fixed the mentioned things and added an explicit diff with the API changes for better text search.

**Review by bcl — APPROVED** [review]
Thanks for the diff.

**Pull request merged**